src/runtimes/docker/clients/DockerClientBase/DockerInfoRecord.ts

```diff
@@ -5,17 +5,30 @@
 
 import { ContainerOS, isContainerOS } from "../../contracts/ContainerClient";
 
+export type DockerPlugins = {
+    Name?: string;
+};
+
+export type DockerClientInfo = {
+    Plugins?: DockerPlugins[];
+};
+
 export type DockerInfoRecord = {
     OperatingSystem?: string;
     OSType?: ContainerOS;
+    ClientInfo?: DockerClientInfo;
 };
-
 export function isDockerInfoRecord(maybeInfo: unknown): maybeInfo is DockerInfoRecord {
+
     const info = maybeInfo as DockerInfoRecord;
 
     if (typeof info.OSType === 'string' && !isContainerOS(info.OSType)) {
         return false;
     }
-
-    return true;
+    else if (!info.ClientInfo) {

```

Comment:
`if (info.ClientInfo && typeof info.ClientInfo !== 'object')` otherwise ClientInfo won't actually be optional.

Reply:
Look at DockerInspectImageRecord.ts for an example of the pattern we've been using for complex record validation.